src/vdbeapi.c

```diff
@@ -1447,7 +1447,12 @@ int sqlite3_bind_blob(
 #ifdef SQLITE_ENABLE_API_ARMOR
   if( nData<0 ) return SQLITE_MISUSE_BKPT;
 #endif
+#ifndef __sqlite_unmodified_upstream
   return bindText(pStmt, i, zData, nData, xDel, 0);
+#else
+  // xDel is a custom deallocator, due to our IT architecture it can't be provided from other modules.
+  return bindText(pStmt, i, zData, nData, free, 0);
+#endif
```

Comment:
I think `ifndef` branches must be swapped because here `free` is used for unmodified sqlite